### PR TITLE
CoinMarketCap: Improve error handling.

### PR DESCRIPTION
**Pull request by sxn**
## Description

Updates API call logic so that errors are reported to the user instead of crashing the extension. While this won't solve the underlying problem, i.e. the fact that the API `403`s, it should at least provide better user experience.

The problem was brought up [here](https://github.com/raycast/extensions/issues/776).

A "proper" fix would be, I guess, updating the extension so that users have to provide an API key. The [free tier API key](https://coinmarketcap.com/api/pricing/) seems to be more than enough for personal use. But I'm not familiar with CoinMarketCap or their API, so I may be off. 😅 

## Type of change

- Bug fix / improvement for somebody else's extension. Extension author: @moojing 

## Screencast

https://user-images.githubusercontent.com/1155589/152328908-502d6bb6-f4be-40da-afca-60ffac8d6505.mov

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)

### CODE REVIEW

File: extensions/coinmarketcap-crypto-crawler/src/index.tsx [hunk 1]
```diff
@@ -190,6 +192,9 @@ export default function SearchCryptoList() {
         refreshExistingCache((err, cryptoList) => {
           if (err) {
             console.error("WriteFileError:" + err);
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            const message = (err as any)?.message;
+            showToast(ToastStyle.Failure, message ? `Refresh failed: ${message}` : "Refresh failed");
```
Comment (mathieudutour) [inline extensions/coinmarketcap-crypto-crawler/src/index.tsx:197]:
```suggestion
            showToast(ToastStyle.Failure, "Refresh failed", message);
```

File: extensions/coinmarketcap-crypto-crawler/src/refresh.tsx [hunk 2]
```diff
@@ -7,7 +7,8 @@ export default async () => {
 
     await showToast(ToastStyle.Success, "Refreshed successfully");
   } catch (error) {
-    console.error(error);
-    await showToast(ToastStyle.Failure, "Refresh failed");
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const message = (error as any)?.message;
+    showToast(ToastStyle.Failure, message ? `Refresh failed: ${message}` : "Refresh failed");
```
Comment (mathieudutour) [inline extensions/coinmarketcap-crypto-crawler/src/refresh.tsx:12]:
```suggestion
    showToast(ToastStyle.Failure, "Refresh failed", message);
```

File: extensions/coinmarketcap-crypto-crawler/src/index.tsx [hunk 3]
```diff
@@ -190,6 +192,9 @@ export default function SearchCryptoList() {
         refreshExistingCache((err, cryptoList) => {
           if (err) {
             console.error("WriteFileError:" + err);
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            const message = (err as any)?.message;
```
Comment (Yonben) [inline extensions/coinmarketcap-crypto-crawler/src/index.tsx:196]: Why isn't the error typed?

Comment (sxn) [reply extensions/coinmarketcap-crypto-crawler/src/index.tsx]: typed it as `Error` now 👍

**Review by mathieudutour — APPROVED** [review]

**Pull request merged**